_includes/_code/lakes-interactive/lib/border-star-button.dart

```diff
@@ -0,0 +1,153 @@
+import 'package:flutter/material.dart';
+// Uncomment lines 3 and 6 to view the visual layout at runtime.
+//import 'package:flutter/rendering.dart' show debugPaintSizeEnabled;
+
+void main() {
+  //debugPaintSizeEnabled = true;
+  runApp(new MyApp());
+}
+
+class FavoriteWidget extends StatefulWidget {
+  @override
+  FavoriteWidgetState createState() => new FavoriteWidgetState();
+}
+
+class FavoriteWidgetState extends State<FavoriteWidget> {
```

Comment:
> The best practice for making a State object

That's fair. I agree that this doc should talk about, and show, best practice for State objects. Maybe we can even add the "best practice" sentence to the doc (unless it's already there :)

@Sfshaza apologies for asking you to change this code in the first place. :(

/me wonders if we need a lint for "make State private and immutable"